File: app/src/main/java/app/trigger/https/IgnoreExpirationTrustManager.java
package app.trigger.https;

import java.math.BigInteger;
import java.security.InvalidKeyException;
import java.security.NoSuchAlgorithmException;
import java.security.NoSuchProviderException;
import java.security.Principal;
import java.security.PublicKey;
import java.security.SignatureException;
import java.security.cert.CertificateEncodingException;
import java.security.cert.CertificateException;
import java.security.cert.X509Certificate;
import java.util.Arrays;
import java.util.Date;
import java.util.Set;

import javax.net.ssl.X509TrustManager;


class IgnoreExpirationTrustManager implements X509TrustManager {
    private final X509TrustManager innerTrustManager;

    public IgnoreExpirationTrustManager(X509TrustManager innerTrustManager) {
        this.innerTrustManager = innerTrustManager;
    }

    @Override
    public void checkClientTrusted(X509Certificate[] chain, String authType) throws CertificateException {
        this.innerTrustManager.checkClientTrusted(chain, authType);
    }

    @Override
    public void checkServerTrusted(X509Certificate[] chain, String authType) throws CertificateException {
        chain = Arrays.copyOf(chain, chain.length);
        X509Certificate[] newChain = new X509Certificate[chain.length];
        newChain[0] = new EternalCertificate(chain[0]);
        System.arraycopy(chain, 1, newChain, 1, chain.length - 1);
        chain = newChain;
    	this.innerTrustManager.checkServerTrusted(chain, authType);
    }

    @Override
    public X509Certificate[] getAcceptedIssuers() {
        return this.innerTrustManager.getAcceptedIssuers();
    }

    private class EternalCertificate extends X509Certificate {
        private final X509Certificate originalCertificate;

        public EternalCertificate(X509Certificate originalCertificate) {
            this.originalCertificate = originalCertificate;
        }

        @Override
        public void checkValidity() {
            // ignore notBefore/notAfter
        }

        @Override
        public void checkValidity(Date date) {
            // ignore notBefore/notAfter
        }

        @Override
        public int getVersion() {
            return originalCertificate.getVersion();
        }

        @Override
        public BigInteger getSerialNumber() {
            return originalCertificate.getSerialNumber();
        }

        @Override
        public Principal getIssuerDN() {
            return originalCertificate.getIssuerDN();
        }

        @Override
        public Principal getSubjectDN() {
            return originalCertificate.getSubjectDN();
        }

        @Override
        public Date getNotBefore() {
            return originalCertificate.getNotBefore();
        }

        @Override
        public Date getNotAfter() {
            return originalCertificate.getNotAfter();
        }

        @Override
        public byte[] getTBSCertificate() throws CertificateEncodingException {
            return originalCertificate.getTBSCertificate();
        }

        @Override
        public byte[] getSignature() {
            return originalCertificate.getSignature();
        }

        @Override
        public String getSigAlgName() {
            return originalCertificate.getSigAlgName();
        }

        @Override
        public String getSigAlgOID() {
            return originalCertificate.getSigAlgOID();
        }

        @Override
        public byte[] getSigAlgParams() {
            return originalCertificate.getSigAlgParams();
        }

        @Override
        public boolean[] getIssuerUniqueID() {
            return originalCertificate.getIssuerUniqueID();
        }

        @Override
        public boolean[] getSubjectUniqueID() {
            return originalCertificate.getSubjectUniqueID();
        }

        @Override
        public boolean[] getKeyUsage() {
            return originalCertificate.getKeyUsage();
        }

        @Override
        public int getBasicConstraints() {
            return originalCertificate.getBasicConstraints();
        }

        @Override
        public byte[] getEncoded() throws CertificateEncodingException {
            return originalCertificate.getEncoded();
        }

        @Override
        public void verify(PublicKey key) throws CertificateException, NoSuchAlgorithmException, InvalidKeyException, NoSuchProviderException, SignatureException {
            originalCertificate.verify(key);
        }

        @Override
        public void verify(PublicKey key, String sigProvider) throws CertificateException, NoSuchAlgorithmException, InvalidKeyException, NoSuchProviderException, SignatureException {
            originalCertificate.verify(key, sigProvider);
        }

        @Override
        public String toString() {
            return originalCertificate.toString();
        }

        @Override
        public PublicKey getPublicKey() {
            return originalCertificate.getPublicKey();
        }

        @Override
        public Set<String> getCriticalExtensionOIDs() {
            return originalCertificate.getCriticalExtensionOIDs();
        }

        @Override
        public byte[] getExtensionValue(String oid) {
            return originalCertificate.getExtensionValue(oid);
        }

        @Override
        public Set<String> getNonCriticalExtensionOIDs() {
            return originalCertificate.getNonCriticalExtensionOIDs();
        }

        @Override
        public boolean hasUnsupportedCriticalExtension() {
            return originalCertificate.hasUnsupportedCriticalExtension();
        }
    }
}

File: app/src/main/java/app/trigger/https/IgnoreExpirationTrustManager.kt
package app.trigger.https

import kotlin.Throws
import java.math.BigInteger
import java.security.*
import java.security.cert.CertificateEncodingException
import java.security.cert.CertificateException
import java.security.cert.X509Certificate
import java.util.*
import javax.net.ssl.X509TrustManager

internal class IgnoreExpirationTrustManager(private val innerTrustManager: X509TrustManager) : X509TrustManager {
    @Throws(CertificateException::class)
    override fun checkClientTrusted(chain: Array<X509Certificate>, authType: String) {
        innerTrustManager.checkClientTrusted(chain, authType)
    }

    @Throws(CertificateException::class)
    override fun checkServerTrusted(chain: Array<X509Certificate>, authType: String) {
        val newChain = arrayOf<X509Certificate>(EternalCertificate(chain[0]))
        /*
        var chain = chain
        chain = Arrays.copyOf(chain, chain.size)
        val newChain = arrayOfNulls<X509Certificate>(chain.size)
        newChain[0] = EternalCertificate(chain[0])
        System.arraycopy(chain, 1, newChain, 1, chain.size - 1)
        chain = newChain
        innerTrustManager.checkServerTrusted(chain, authType)
        */
        innerTrustManager.checkServerTrusted(newChain, authType)
    }

    override fun getAcceptedIssuers(): Array<X509Certificate> {
        return innerTrustManager.acceptedIssuers
    }

    private inner class EternalCertificate(private val originalCertificate: X509Certificate) : X509Certificate() {
        override fun checkValidity() {
            // ignore notBefore/notAfter
        }

        override fun checkValidity(date: Date) {
            // ignore notBefore/notAfter
        }

        override fun getVersion(): Int {
            return originalCertificate.version
        }

        override fun getSerialNumber(): BigInteger {
            return originalCertificate.serialNumber
        }

        override fun getIssuerDN(): Principal {
            return originalCertificate.issuerDN
        }

        override fun getSubjectDN(): Principal {
            return originalCertificate.subjectDN
        }

        override fun getNotBefore(): Date {
            return originalCertificate.notBefore
        }

        override fun getNotAfter(): Date {
            return originalCertificate.notAfter
        }

        @Throws(CertificateEncodingException::class)
        override fun getTBSCertificate(): ByteArray {
            return originalCertificate.tbsCertificate
        }

        override fun getSignature(): ByteArray {
            return originalCertificate.signature
        }

        override fun getSigAlgName(): String {
            return originalCertificate.sigAlgName
        }

        override fun getSigAlgOID(): String {
            return originalCertificate.sigAlgOID
        }

        override fun getSigAlgParams(): ByteArray {
            return originalCertificate.sigAlgParams
        }

        override fun getIssuerUniqueID(): BooleanArray {
            return originalCertificate.issuerUniqueID
        }

        override fun getSubjectUniqueID(): BooleanArray {
            return originalCertificate.subjectUniqueID
        }

        override fun getKeyUsage(): BooleanArray {
            return originalCertificate.keyUsage
        }

        override fun getBasicConstraints(): Int {
            return originalCertificate.basicConstraints
        }

        @Throws(CertificateEncodingException::class)
        override fun getEncoded(): ByteArray {
            return originalCertificate.encoded
        }

        @Throws(CertificateException::class, NoSuchAlgorithmException::class, InvalidKeyException::class, NoSuchProviderException::class, SignatureException::class)
        override fun verify(key: PublicKey) {
            originalCertificate.verify(key)
        }

        @Throws(CertificateException::class, NoSuchAlgorithmException::class, InvalidKeyException::class, NoSuchProviderException::class, SignatureException::class)
        override fun verify(key: PublicKey, sigProvider: String) {
            originalCertificate.verify(key, sigProvider)
        }

        override fun toString(): String {
            return originalCertificate.toString()
        }

        override fun getPublicKey(): PublicKey {
            return originalCertificate.publicKey
        }

        override fun getCriticalExtensionOIDs(): Set<String> {
            return originalCertificate.criticalExtensionOIDs
        }

        override fun getExtensionValue(oid: String): ByteArray {
            return originalCertificate.getExtensionValue(oid)
        }

        override fun getNonCriticalExtensionOIDs(): Set<String> {
            return originalCertificate.nonCriticalExtensionOIDs
        }

        override fun hasUnsupportedCriticalExtension(): Boolean {
            return originalCertificate.hasUnsupportedCriticalExtension()
        }
    }
}